NHWC issue on YoloX

```
Running [ MIGraphX Version: 2.12.0. ]: ./bin/driver perf /data/yolox_l_dyn.onnx --fp16
MIGRAPHX_DISABLE_MIOPEN_POOLING=1
MIGRAPHX_ENABLE_NHWC=1
MIGRAPHX_MLIR_USE_SPECIFIC_OPS=convolution,dot,fused
MIGRAPHX_DISABLE_LAYERNORM_FUSION=1
Compiling ... 
Reading: /data/yolox_l_dyn.onnx
main.cpp:25:47: error: use of overloaded operator '+' is ambiguous (with operand types 'float' and 'decltype(math::wrap(math::wrap_exp, xs))' (aka '__half'))
   25 | auto zz3 = migraphx::convert<half>(1.f / (1.f + migraphx::exp(-zz2)));
      |                                           ~~~ ^ ~~~~~~~~~~~~~~~~~~~
main.cpp:48:94: note: in instantiation of function template specialization 'migraphx::pointwise1<_Float16, _Float16>' requested here
   48 |         concat<3>(pack(MIGRAPHX_LIFT(pointwise0), pointwise00_concat_x0), pack(MIGRAPHX_LIFT(pointwise1), pointwise11_concat_x0, pointwise11_concat_x1))(MIGRAPHX_LIFT(pointwise2), y, xs...);
      |                                                                                              ^
./migraphx/kernels/concat.hpp:68:54: note: while substituting deduced template arguments into function template 'operator()' [with private_lifts_xs:auto = <_Float16 &, _Float16 &>]
   68 |                               [&](auto i) { z[i] = f(g(x[i], xs[i]...), ys[i]...); });
      |                                                      ^
./migraphx/kernels/index.hpp:200:64: note: in instantiation of function template specialization 'migraphx::concat_each(index, migraphx::integral_constant<unsigned int, 4>, (lambda at ./migraphx/kernels/functional.hpp:273:12), (lambda at main.cpp:48:154), migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 80, 80, 85>, migraphx::integral_const_array<unsigned int, 544000, 6800, 85, 1>>>)::(anonymous class)::operator()((lambda at main.cpp:48:80), migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 80, 80, 81>, migraphx::integral_const_array<unsigned int, 518400, 6480, 81, 1>>>, migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 80, 80, 81>, migraphx::integral_const_array<unsigned int, 0, 0, 0, 1>>>)::(anonymous class)::operator()(migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 1, 80, 80, 81>, migraphx::integral_const_array<unsigned int, 544000, 6800, 85, 1>>>)::(anonymous class)::operator()<unsigned int>' requested here
  200 |     static constexpr auto invoke_loop(F f, I i, D) -> decltype(f(i))
      |                                                                ^
./migraphx/kernels/index.hpp:224:13: note: while substituting deduced template arguments into function template 'invoke_loop' [with F = (lambda at ./migraphx/kernels/concat.hpp:68:31), I = index_int, D = index_int]
  224 |             invoke_loop(f, i, k);
      |             ^
./migraphx/kernels/index.hpp:256:17: note: in instantiation of function template specialization 'migraphx::index::for_stride_loop<(lambda at ./migraphx/kernels/concat.hpp:68:31), migraphx::integral_constant<unsigned int, 518400>, migraphx::integral_constant<unsigned int, 272000>>' requested here
  256 |                 for_stride_loop(start, n, stride, f);
      |                 ^
./migraphx/kernels/index.hpp:268:9: note: in instantiation of function template specialization 'migraphx::index::for_stride<false, (lambda at ./migraphx/kernels/concat.hpp:68:31), migraphx::integral_constant<unsigned int, 518400>, migraphx::integral_constant<unsigned int, 272000>>' requested here
  268 |         for_stride<false>(global, n, nglobal(), f);
      |         ^
./migraphx/kernels/concat.hpp:67:17: note: in instantiation of function template specialization 'migraphx::index::global_stride<(lambda at ./migraphx/kernels/concat.hpp:68:31), migraphx::integral_constant<unsigned int, 518400>>' requested here
   67 |             idx.global_stride(x.get_shape().elements(),
      |                 ^
main.cpp:25:47: note: built-in candidate operator+(float, float)
   25 | auto zz3 = migraphx::convert<half>(1.f / (1.f + migraphx::exp(-zz2)));
      |                                               ^
```

Experiencing the same issue here:

`Running [ MIGraphX Version: 2.12.0.ebf82f61f ]: ./build/bin/migraphx-driver perf owlvit-base-patch32/model.onnx --fp16 --batch 1 --input-dim @pixel_values 1 3 768 768 --fill1 input_ids
MIGRAPHX_ENABLE_NHWC=1
MIGRAPHX_MLIR_USE_SPECIFIC_OPS=convolution,fused_convolution
Compiling ... 
Reading: owlvit-base-patch32/model.onnx
main.cpp:14:87: error: use of overloaded operator '-' is ambiguous (with operand types 'decltype(math::wrap(math::wrap_exp, xs))' (aka '__half') and 'int')
   14 | auto zz2 = migraphx::convert<half>(migraphx::where(x0 > 0, x0, 1 * (migraphx::exp(x0) - 1)));
      |                                                                     ~~~~~~~~~~~~~~~~~ ^ ~
main.cpp:41:81: note: in instantiation of function template specialization 'migraphx::inner_pointwise<_Float16, _Float16, _Float16, bool>' requested here
   41 |     pointwise<1, false>(idx, transform_args(), vectorize<1, 0>())(MIGRAPHX_LIFT(inner_pointwise), private_p0,private_p1,private_p2,private_p3,private_p4);
      |                                                                                 ^
./migraphx/kernels/pointwise.hpp:42:18: note: while substituting deduced template arguments into function template 'operator()' [with private_lifts_xs:auto = <_Float16 &, _Float16 &, _Float16 &, bool &>]
   42 |         auto r = f(x[i], xs[i]...);
      |                  ^
./migraphx/kernels/index.hpp:200:64: note: in instantiation of function template specialization 'migraphx::pointwise_tensor((lambda at ./migraphx/kernels/tile.hpp:148:16), (lambda at main.cpp:41:67), (lambda at ./migraphx/kernels/functional.hpp:273:12), migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 576>, migraphx::integral_const_array<unsigned int, 1282>>>, migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 576>, migraphx::integral_const_array<unsigned int, 1>>>, migraphx::tensor_view<_Float16, migraphx::shape<migraphx::integral_const_array<unsigned int, 576>, migraphx::integral_const_array<unsigned int, 1282>>>, migraphx::tensor_view<bool, migraphx::shape<migraphx::integral_const_array<unsigned int, 576>, migraphx::integral_const_array<unsigned int, 0>>>)::(anonymous class)::operator()<unsigned int>' requested here
  200 |     static constexpr auto invoke_loop(F f, I i, D) -> decltype(f(i))
      |                                                                ^
./migraphx/kernels/index.hpp:244:21: note: while substituting deduced template arguments into function template 'invoke_loop' [with F = (lambda at ./migraphx/kernels/pointwise.hpp:41:38), I = index_int, D = integral_constant<decltype(0), 0>]
  244 |                     invoke_loop(f, start, _c<0>);
      |                     ^
./migraphx/kernels/index.hpp:268:9: note: in instantiation of function template specialization 'migraphx::index::for_stride<false, (lambda at ./migraphx/kernels/pointwise.hpp:41:38), migraphx::integral_constant<unsigned int, 576>, migraphx::integral_constant<unsigned int, 576>>' requested here
  268 |         for_stride<false>(global, n, nglobal(), f);`

--- a/src/targets/gpu/kernels/include/migraphx/kernels/math.hpp
+++ b/src/targets/gpu/kernels/include/migraphx/kernels/math.hpp
@@ -31,6 +31,7 @@
 #include <migraphx/kernels/hip.hpp>
 #include <migraphx/kernels/float8.hpp>
 #include <migraphx/kernels/pp.hpp>
+#include <migraphx/kernels/bit_cast.hpp>
 
 namespace migraphx {
 
@@ -45,6 +46,17 @@ constexpr auto as_float(T x)
         return float(x);
 }
 
+template <class T>
+constexpr auto to_native(T x)
+{
+    return x;
+}
+
+constexpr migraphx::half to_native(__half x)
+{
+    return bit_cast<migraphx::half>(x);
+}
+
 template <class F, class T, class... Ts, MIGRAPHX_REQUIRES(not is_any_vec<T, Ts...>())>
 __device__ auto wrap(F f, T x, Ts... xs)
 {
@@ -54,7 +66,7 @@ __device__ auto wrap(F f, T x, Ts... xs)
     }
     else if constexpr(is_callable<F, T, Ts...>{})
     {
-        return f(x, xs...);
+        return to_native(f(x, xs...));
     }
     else
     {

--- a/src/targets/gpu/kernels/include/migraphx/kernels/math.hpp
+++ b/src/targets/gpu/kernels/include/migraphx/kernels/math.hpp
@@ -52,10 +52,7 @@ constexpr auto to_native(T x)
     return x;
 }
 
-constexpr migraphx::half to_native(__half x)
-{
-    return bit_cast<migraphx::half>(x);
-}
+constexpr migraphx::half to_native(__half x) { return bit_cast<migraphx::half>(x); }
 
 template <class F, class T, class... Ts, MIGRAPHX_REQUIRES(not is_any_vec<T, Ts...>())>
 __device__ auto wrap(F f, T x, Ts... xs)

--- a/test/gpu/jit.cpp
+++ b/test/gpu/jit.cpp
@@ -181,7 +181,7 @@ __global__ void kernel(${type}* p)
 {
     auto x = *p;
     *p = migraphx::test_implicit_conversion(migraphx::${invoke});
-
+    (void)1.f + migraphx::vec_at(migraphx::${invoke}, 0);
 }
 }
 }

--- a/test/gpu/jit.cpp
+++ b/test/gpu/jit.cpp
@@ -181,7 +181,7 @@ __global__ void kernel(${type}* p)
 {
     auto x = *p;
     *p = migraphx::test_implicit_conversion(migraphx::${invoke});
-    (void)1.f + migraphx::vec_at(migraphx::${invoke}, 0);
+    (void)(1.f + migraphx::vec_at(migraphx::${invoke}, 0));
 }
 }
 }

